compiler/src/iree/compiler/Codegen/Common/GPU/GPUPackToIntrinsics.cpp

```diff
@@ -31,37 +32,59 @@ struct GPUPackToIntrinsicsPass final
 };
 } // namespace
 
-LogicalResult packToIntrinsic(linalg::LinalgOp linalgOp,
-                              RewriterBase &rewriter) {
-  auto loweringConfig =
-      getLoweringConfig<IREE::GPU::LoweringConfigAttr>(linalgOp);
-  assert(loweringConfig && "Packing unconfigured op");
+FailureOr<SmallVector<OpFoldResult>>
+getPackedSizes(linalg::LinalgOp linalgOp, RewriterBase &rewriter,
+               IREE::Codegen::InnerTileDescAttrInterface kind) {
+  SmallVector<int64_t> dims;
+  SmallVector<SmallVector<unsigned, 2>> indices;
+  auto createPackedSizes =
+      [&rewriter, &linalgOp](SmallVector<int64_t> dims,
+                             SmallVector<SmallVector<unsigned, 2>> indices)
+      -> FailureOr<SmallVector<OpFoldResult>> {
+    auto zero = rewriter.getIndexAttr(0);
+    SmallVector<OpFoldResult> packedSizes(linalgOp.getNumLoops(), zero);
+    for (auto [dim, index] : llvm::zip_equal(dims, indices)) {
+      if (index.empty()) {
+        linalgOp.emitError()
+            << "contraction like operation missing critical dimension\n";
+        return failure();
+      }
+      packedSizes[index.back()] = rewriter.getIndexAttr(dim);
+    }
+    return packedSizes;
+  };
 
-  IREE::GPU::MmaInterfaceAttr kind = getMmaKind(loweringConfig);
-  assert(kind && "Packing op without mma kind");
+  FailureOr<IREE::LinalgExt::ScaledContractionDimensions> scaledContrDims =
+      IREE::LinalgExt::inferScaledContractionDims(linalgOp);
+  if (succeeded(scaledContrDims)) {
+    auto [m, n, k, kB] = IREE::GPU::ScaledMMAAttr::getScaledMNKShape(kind);
+    indices = {scaledContrDims->m, scaledContrDims->n, scaledContrDims->k,
+               scaledContrDims->kB};
+    return createPackedSizes({m, n, k, kB}, indices);
+  }
 
   FailureOr<linalg::ContractionDimensions> contractionDims =
       linalg::inferContractionDims(linalgOp);
   if (failed(contractionDims)) {
-    return rewriter.notifyMatchFailure(linalgOp,
-                                       "failed to infer contraction dims");
+    linalgOp.emitError() << "failed to infer contraction dims\n";
```

Comment:
... Turn this back to a notifyMatchFailure(); llvm::errs() prints should never land